src/zernike_fet.cpp

```diff
@@ -0,0 +1,291 @@
+    // First write coordinates and number of groups
+    std::vector<double> x_bounds(Nx + 1, 0.);
+    for (std::size_t i = 0; i <= Nx; i++) {
+      x_bounds[i] = (static_cast<double>(i) * dx) + r_low.x();
+    }
+    tally_grp.createAttribute("x-bounds", x_bounds);
+
+    std::vector<double> y_bounds(Ny + 1, 0.);
+    for (std::size_t i = 0; i <= Ny; i++) {
+      y_bounds[i] = (static_cast<double>(i) * dy) + r_low.y();
+    }
+    tally_grp.createAttribute("y-bounds", y_bounds);
+
+    std::vector<double> z_bounds(Nz + 1, 0.);
+    for (std::size_t i = 0; i <= Nz; i++) {
+      z_bounds[i] = (static_cast<double>(i) * dz) + r_low.z();
+    }
+    tally_grp.createAttribute("z-bounds", z_bounds);
+
+
+    tally_grp.createAttribute("energy-bounds", energy_bounds);
+  */
+  // Save the quantity
+  tally_grp.createAttribute("quantity", quantity_str());
+
+  /*if (this->quantity_str() == "mt") {
+    tally_grp.createAttribute("mt", this->mt());
+  }*/
+
+  // Save the estimator
+  tally_grp.createAttribute("estimator", estimator_str());
+
+  // Convert flux_var to the error on the mean
+  for (size_t l = 0; l < tally_var_.size(); l++)
+    tally_var_[l] = std::sqrt(tally_var_[l] / static_cast<double>(gen_));
+
+  // Add data sets for the average and the standard deviation
+  auto avg_dset =
+      tally_grp.createDataSet<double>("avg", H5::DataSpace(tally_avg_.shape()));
+  avg_dset.write_raw(&tally_avg_[0]);
+
+  auto std_dset =
+      tally_grp.createDataSet<double>("std", H5::DataSpace(tally_var_.shape()));
+  std_dset.write_raw(&tally_var_[0]);
+}
+
+// make the tally-zernike
+std::shared_ptr<ZernikeFET> make_zernike_fet(const YAML::Node& node) {
+  // Check the name of the tallt is given or not.
+  if (!node["name"] && !node["name"].IsScalar()) {
+    fatal_error("No valid name is provided.");
+  }
+  std::string zernike_fet_tally_name = node["name"].as<std::string>();
+
+  // Check the estimator is given or not.
+  if (!node["estimator"] && !node["estimator"].IsScalar()) {
+    fatal_error("No valid estimator is given for " + zernike_fet_tally_name +
+                " tally.");
+  }
+  std::string estimator_name = node["estimator"].as<std::string>();
+
+  // Check for the quantity
+  std::string given_quantity = "";
+  if (!node["quantity"] && !node["quantity"].IsScalar()) {
+    fatal_error("No valid quantity is given for " + zernike_fet_tally_name +
+                " tally.");
+  }
+  given_quantity = node["quantity"].as<std::string>();
+  ZernikeFET::Quantity quant;
+
+  if (given_quantity == "flux") {
+    quant = ZernikeFET::Quantity::Flux;
+  } else if (given_quantity == "fission") {
+    quant = ZernikeFET::Quantity::Fission;
+  } else if (given_quantity == "absorption") {
+    quant = ZernikeFET::Quantity::Absorption;
+  } else if (given_quantity == "elastic") {
+    quant = ZernikeFET::Quantity::Elastic;
+  } else {
+    fatal_error("For " + zernike_fet_tally_name +
+                " tally, a unknown quantity is given.");
+  }
```

Comment:
We do this all the time. How about we add a function in `itally.hpp/cpp` which reads the quantity enum for us. Then we don't need to re-write this all the time.